src/linkPanel/widgets/linkedDataset.ts

```diff
@@ -66,40 +26,60 @@ export class LinkedDataset extends LinkEditorWidget {
       this._createdLinksView.widgets[0].dispose();
     }
 
-    // Get a set of the linked dataset.
+    // Get a set of the identity linked dataset.
     const datasetLinks = new Set<string>();
-    Array.from(this._linkEditorModel.relatedLinks.values()).map(relatedLink => {
-      if (relatedLink.src && relatedLink.dest) {
+    Array.from(this._linkEditorModel.relatedLinks.values()).forEach(
+      relatedLink => {
+        if (relatedLink.src && relatedLink.dest) {
+          datasetLinks.add(
+            JSON.stringify(
```

Comment:
Why do we store a string instead of an array?

Reply:
We need here to keep unique sets of datasets, even if there are several links for the same set of datasets.
The string is easier to use to ensure uniqueness.